File: java/timesheet/src/test/java/timesheet/FreelancerBuilder.java
package timesheet;

import java.time.LocalDate;

public class FreelancerBuilder {
    private String lastName = "Doe";
    private String email = "john.doe@foobar.com";

    public static FreelancerBuilder aFreelancer(){
        return new FreelancerBuilder();
    }

    public FreelancerBuilder withEmail(String email) {
        this.email = email;
        return this;
    }


    public FreelancerBuilder withLastName(String lastName) {
        this.lastName = lastName;
        return this;
    }

    public Freelancer build(){
        return new Freelancer(lastName, email);
    }
}

File: scala/timesheet/src/test/scala/timesheet/FreelancerBuilder.scala
package timesheet

class FreelancerBuilder {
  private var lastName: String = "Doe"
  private var email: String = "john.doe@foobar.com"

  def withEmail(email: String): FreelancerBuilder = {
    this.email = email
    this
  }

  def withLastName(lastName: String): FreelancerBuilder = {
    this.lastName = lastName
    this
  }

  def build(): Worker = {
    new Freelancer(lastName, email)
  }
}

object FreelancerBuilder {
  def aFreelancer(): FreelancerBuilder = new FreelancerBuilder()
}
